### DISCUSSION TITLE
Is there any possible to use index outside the return

### DISCUSSION THREAD

**Discussion by abdulmateentechbits** [react-hook-form/react-hook-form#9190] [Q&A] [answered]
How can i use index out side of the return for useWatch to capture the nested element
`{
fields.map((item, index) => {
          return (
            <li key={item.id}>
              <label> single input </label>
              <input
                name={`test[${index}].task`}
                ref={register()}
                defaultValue={item.task}
              />
    )}
)}`

**Comment by bluebill1049** [comment, marked as answer]
i guess similar question: https://github.com/react-hook-form/react-hook-form/discussions/9196
